<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1190</width>
    <height>797</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="pushBackButton">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>50</y>
      <width>221</width>
      <height>101</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>25</pointsize>
     </font>
    </property>
    <property name="text">
     <string>push_Back</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushFrontButton">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>50</y>
      <width>221</width>
      <height>101</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>25</pointsize>
     </font>
    </property>
    <property name="text">
     <string>push_Front</string>
    </property>
   </widget>
   <widget class="QPushButton" name="popBackButton">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>50</y>
      <width>221</width>
      <height>101</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>25</pointsize>
     </font>
    </property>
    <property name="text">
     <string>pop_Back</string>
    </property>
   </widget>
   <widget class="QPushButton" name="popFrontButton">
    <property name="geometry">
     <rect>
      <x>770</x>
      <y>50</y>
      <width>221</width>
      <height>101</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>25</pointsize>
     </font>
    </property>
    <property name="text">
     <string>pop_Front</string>
    </property>
   </widget>
   <widget class="QListWidget" name="listWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>1111</width>
      <height>691</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="valueInput">
    <property name="geometry">
     <rect>
      <x>50</x>
      <y>170</y>
      <width>281</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>21</pointsize>
     </font>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
    <property name="placeholderText">
     <string>введите элемент</string>
    </property>
   </widget>
   <widget class="QTextEdit" name="dequeDisplay">
    <property name="geometry">
     <rect>
      <x>60</x>
      <y>260</y>
      <width>931</width>
      <height>421</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>23</pointsize>
     </font>
    </property>
   </widget>
   <widget class="QLabel" name="sizeLabel">
    <property name="geometry">
     <rect>
      <x>590</x>
      <y>170</y>
      <width>131</width>
      <height>70</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>21</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Size</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QLabel" name="emptyLabel">
    <property name="geometry">
     <rect>
      <x>820</x>
      <y>170</y>
      <width>131</width>
      <height>70</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>21</pointsize>
     </font>
    </property>
    <property name="text">
     <string>empty</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="clearButton">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>170</y>
      <width>221</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>25</pointsize>
     </font>
    </property>
    <property name="text">
     <string>clear</string>
    </property>
   </widget>
   <zorder>listWidget</zorder>
   <zorder>pushBackButton</zorder>
   <zorder>pushFrontButton</zorder>
   <zorder>popBackButton</zorder>
   <zorder>popFrontButton</zorder>
   <zorder>valueInput</zorder>
   <zorder>dequeDisplay</zorder>
   <zorder>sizeLabel</zorder>
   <zorder>emptyLabel</zorder>
   <zorder>clearButton</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1190</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
